### ISSUE TITLE
halo mass unit is not consistent with schema (for cosmoDC2, and maybe protoDC2)

### ISSUE THREAD

**Issue by yymao** [LSSTDESC/gcr-catalogs#285]
The unit halo mass in cosmoDC2 is currently Msun/h, but it should be Msun according to the schema. 

This can be easily fixed by updating the quantity modifier in the reader. @evevkovacs will help make sure that the unit is consistent in both cosmoDC2 and protoDC2 (or apply fixes if not). 

Before this is fixed, users can edit the quantity modifier directly. This way, users will not need to update their code again when the fix is applied to the reader. For example, for cosmoDC2,

```python
catalog.add_quantity_modifier(
    'halo_mass', 
    (lambda x: x / catalog.cosmology.h, 'hostHaloMass'), 
    overwrite=True
)
```

Thanks to @marina-ricci for discovering this. Thanks to @katrinheitmann and @evevkovacs for identifying the issue.

**Comment by evevkovacs** [comment]
@yymao I think my preferred fix would be to correct the units for hostHaloMass (native quantity) to be M_sun/h and then modify the computation of 'haloMass' (schema quantity) to include the factor of h and make sure that the units are M_sun. Do you see any problem with handling it this way?

**Comment by yymao** [comment]
@evevkovacs I think I agree with you if my interpretation of what you say is correct: 
1. By "correct the units for `hostHaloMass`", you just mean correct the quantity information text, not the actual data units, right? I believe `hostHaloMass` is already in Msun/h
2. By "the computation of `halo_mass`" you mean the quantity modifier, right?

If yes to both, then, yes, I think this is how we should handle it.

**Comment by evevkovacs** [comment]
@yymao Yes to both. Sorry if I was unclear.

**Comment by yymao** [comment]
cosmoDC2 and baseDC2 have been fixed by #286 and #284. protoDC2 remains to be checked.
